### PR TITLE
Convert build system to Perl 5.10.

### PR DESCRIPTION
**Pull request by SaberUK**
Now we have dropped support for the old CentOS release we can use Perl 5.10 instead of 5.8. Yay!

**Note:** I'm not going to update `make/utilities.pm` as that file is scheduled to be removed soon anyway. Similarly, `./modulemanager` and the module bits of `./configure` are due to be entirely rewritten so i'm not going to update that either.

### CODE REVIEW

**Comment by SaberUK** [comment]
@attilamolnar This should be ready to merge now.

**Comment by attilamolnar** [comment]
Thanks, can you rephrase "SSL generation" in the future?
